mass_flask_webui/views/sample.py

```diff
@@ -30,6 +31,27 @@ def sample_detail(sample_id):
             flash('Your comment has been added', 'success')
             return redirect(url_for('.sample_detail', sample_id=sample.id))
         reports = Report.objects(sample=sample)
+
+        analysis_systems = []
+        for system in AnalysisSystem.objects:
+            analysis_systems.append((system.identifier_name, system.verbose_name))
+
+        request_form = RequestAnalysisForm()
+        request_form.analysis_system.choices = analysis_systems
+
+        if request_form.validate_on_submit():
```

Comment:
Additionally I noticed that there seem to be no checks if the user is logged in.